### PR TITLE
[KERNEL32] Add doxygen to all public WINAPI

### PR DESCRIPTION
**Pull request by gunmetal313**
[KERNEL32] Add doxygen to all public WINAPI

I added comments for GetNumaAvailableMemoryNode. After review, I will do the same for the rest of the procedures in KERNEL32

working in progress

### CODE REVIEW

File: dll/win32/kernel32/client/sysinfo.c [hunk 1]
```diff
@@ -380,13 +380,36 @@ GetNumaProcessorNode(IN UCHAR Processor,
...
+ *        The number of the node.
+ *
+ * @param AvailableBytes
+ *        The amount of available memory for the node, in bytes.
+ *
+ * @return If the function succeeds, the return value is non zero.
+ *         If the function fails, the return value is zero. To get
+ *         extended error information, call GetLastError.
+ *
+ * @remarks The GetNumaAvailableMemoryNode function returns the amount of memory consumed
```
Comment (ThFabba) [inline dll/win32/kernel32/client/sysinfo.c]: These comments use the exact same wording as MSDN. This is not acceptable, those texts are under copyright. You'll have to describe the function in your own words.

File: dll/win32/kernel32/client/sysinfo.c [hunk 2]
```diff
@@ -393,17 +393,8 @@ GetNumaProcessorNode(IN UCHAR Processor,
  * @param AvailableBytes
  *        The amount of available memory for the node, in bytes.
  *
- * @return If the function succeeds, the return value is non zero.
- *         If the function fails, the return value is zero. To get
- *         extended error information, call GetLastError.
+ * @return TRUE or FALSE on error.You can get extended information using the GetLastError function.
```
Comment (MailYouLater) [inline dll/win32/kernel32/client/sysinfo.c]: Wait, what does this function actually return? TRUE/FALSE, or non-zero/zero?
Assuming the previous text was correct, I recommend something more like this.
```
 * @return Returns non-zero value when the function succeeds, or zero when it fails.
           When an error occurs, more information can be retrieved by calling GetLastError.
```

File: dll/win32/kernel32/client/sysinfo.c [hunk 3]
```diff
@@ -393,17 +393,8 @@ GetNumaProcessorNode(IN UCHAR Processor,
...
  *
- * @remarks The GetNumaAvailableMemoryNode function returns the amount of memory consumed
```
Comment (MailYouLater) [inline dll/win32/kernel32/client/sysinfo.c]: I'm not sure what the parts about the memory performance counters mean, so I didn't bother with them, but we should mention that single node systems and multi-node systems are calculated differently.
```
  * @remarks GetNumaAvailableMemoryNode returns how much memory is in free and
             zeroed pages on the node. If a systems has only one node, the
             return value will include standby pages. If a system has more than
             one node, the return value will not include standby pages.
```

File: dll/win32/kernel32/client/sysinfo.c (hunk 2 above)
Comment (Fraizeraust) [reply dll/win32/kernel32/client/sysinfo.c]: `BOOL` is defined as an integer, *int*, with FALSE being 0 whereas TRUE being 1. The TRUE/FALSE wording can be kept however the sentence has to be improved explaining what TRUE and FALSE actually represent (*TRUE or FALSE on error* is incorrect because the function returns FALSE if it fails, it cannot return an either value).

File: dll/win32/kernel32/client/sysinfo.c [hunk 4]
```diff
@@ -380,13 +380,27 @@ GetNumaProcessorNode(IN UCHAR Processor,
...
+ * @return TRUE or FALSE on error.You can get extended information using the GetLastError function.
+ *
+ **/
 BOOL
 WINAPI
-GetNumaAvailableMemoryNode(IN UCHAR Node,
-                           OUT PULONGLONG AvailableBytes)
+GetNumaAvailableMemoryNode(
+    IN UCHAR Node,
+    OUT PULONGLONG AvailableBytes)
```
Comment (Fraizeraust) [inline dll/win32/kernel32/client/sysinfo.c]: Why this change?

File: dll/win32/kernel32/client/sysinfo.c (hunk 2 above)
Comment (MailYouLater) [reply dll/win32/kernel32/client/sysinfo.c]: @Fraizeraust According to MSDN, it's not returning `TRUE` or `FALSE`, or `1` or `0`, it's returning 'a non-zero number' (that represents how much memory is in free and zeroed pages in the node specified), or `0`.  I doubt we want our documentation to incorrectly indicate that upon success it returns `1` or `TRUE`, when it's far more likely to be some other non-zero number most of the time.

Comment (Fraizeraust) [reply dll/win32/kernel32/client/sysinfo.c]: Clearly you haven't looked at the function's prototype, did you?
```C
BOOL
WINAPI
GetNumaAvailableMemoryNode(IN UCHAR Node,
                           OUT PULONGLONG AvailableBytes)
```
And this is how `BOOL` is defined in WinDef.h:
```C
#ifndef FALSE
#define FALSE               0
#endif

#ifndef TRUE
#define TRUE                1
#endif

typedef int                 BOOL;
```
The function's code is also pretty self-explanatory about what it does and returns, really.

Comment (MailYouLater) [reply dll/win32/kernel32/client/sysinfo.c]: Huh, you're right, the prototype does say `BOOL`, so is MSDN wrong where it says "The GetNumaAvailableMemoryNode function returns the amount of memory consumed by free and zeroed pages on the specified node." or is it supposed to 'return' that value by some other means?

Comment (MailYouLater) [reply dll/win32/kernel32/client/sysinfo.c]: Nevermind. Actually looking at the code in the ReactOS implementation, it's C code, and one parameter is IN, and the other is OUT, and that's how it's 'returning' the amount while actually `return`ing `TRUE`.

derp

File: dll/win32/kernel32/client/sysinfo.c [hunk 5]
```diff
@@ -380,13 +380,27 @@ GetNumaProcessorNode(IN UCHAR Processor,
...
 
-/*
- * @implemented
- */
+/**
+ *
+ * @name GetNumaAvailableMemoryNode
+ * @implemented NT 5.2
+ *
+ *     Retrieves the amount of memory available in the specified node.
```
Comment (ThFabba) [inline dll/win32/kernel32/client/sysinfo.c]: It's better after your changes, but I'd appreciate if the function and parameter descriptions were also changed at least slightly.

File: dll/win32/kernel32/client/sysinfo.c (hunk 4 above)
Comment (MailYouLater) [reply dll/win32/kernel32/client/sysinfo.c]: As ThFabba says, keeping a PR in scope will make it easier to review, and reduces the likelyhood of unnecessary discussion which can prolong an issue.  Also, I'm not sure where, but I remember reading a recommendation that if you make style only changes to a file, they should be in a separate commit from functional changes.
